Common/cpp/SharedWorkletStarter.cpp

```diff
@@ -26,7 +35,93 @@ SharedWorkletStarter::~SharedWorkletStarter() {
 }
 
 jsi::Value SharedWorkletStarter::asParameter(jsi::Runtime &rt) {
-  return jsi::Value(this->id);
+  
+  class HO : public jsi::HostObject {
+    public:
+    int id;
+    SharedWorkletStarter *starter;
```

Comment:
Why not std::shared_ptr ?

Reply:
because `this` is passed down to the inner class